tests/test_blocked_well.py

```diff
@@ -0,0 +1,588 @@
+                else:
+                    fp.write(f' {row[col]:6.2f}')
+            fp.write('\n')
+    bw = resqpy.well.BlockedWell(
+        model,
+        wellspec_file = wellspec_file,
+        well_name = well_name,
+        use_face_centres = True,
+        add_wellspec_properties = True,
+    )
+    assert bw is not None
+    bw_uuid = bw.uuid
+    skin_uuid = model.uuid(title = 'SKIN', related_uuid = bw.uuid)
+    assert skin_uuid is not None
+    skin_prop = rqp.Property(model, uuid = skin_uuid)
+    assert skin_prop is not None
+    assert_array_almost_equal(skin_prop.array_ref(), [0.0, 2.5, 1.0, -0.5])
+    model.store_epc()
+    print(model.grid().property_collection.titles())
+    # re-open model from persistent storage
+    model = Model(model.epc_file)
+    bw2_uuid = model.uuid(obj_type = 'BlockedWellboreRepresentation', title = 'DOGLEG')
+    assert bw2_uuid is not None
+    bw2 = resqpy.well.BlockedWell(model, uuid = bw2_uuid)
+    assert bu.matching_uuids(bw_uuid, bw2_uuid)
+    df2 = bw.dataframe(extra_columns_list = ['ANGLV', 'ANGLA', 'LENGTH', 'SKIN', 'RADW', 'PPERF'],
+                       use_properties = True)
+    assert df2 is not None
+    assert len(df2.columns) == 9
+    for col in ['ANGLV', 'ANGLA', 'SKIN', 'RADW', 'PPERF']:
+        assert_array_almost_equal(np.array(source_df[col]), np.array(df2[col]))
+    df3 = bw.dataframe(extra_columns_list = ['ANGLV', 'ANGLA', 'LENGTH', 'SKIN', 'RADW', 'PPERF'],
+                       use_properties = ['SKIN', 'RADW'],
+                       perforation_list = [(125, 175)])
+    for col in ['SKIN', 'RADW']:
+        assert_array_almost_equal(np.array(source_df[col]), np.array(df3[col]))
+
+
+@pytest.mark.parametrize('check_grid_name,name_for_check,col_list', [(True, 'BATTLESHIP', ['IW', 'JW', 'L', 'GRID']),
+                                                                     (False, None, ['IW', 'JW', 'L'])])
+def test_derive_from_wellspec_check_grid_name(example_model_and_crs, check_grid_name, name_for_check, col_list):
+
+    # --------- Arrange ----------
+    model, crs = example_model_and_crs
+    grid = RegularGrid(model,
+                       extent_kji = (5, 3, 3),
+                       dxyz = (50.0, -50.0, 50.0),
+                       origin = (0.0, 0.0, 100.0),
+                       crs_uuid = crs.uuid,
+                       title = 'Battleship',
+                       set_points_cached = True)
+    grid.write_hdf5()
+    grid.create_xml(write_geometry = True)
+    well_name = 'DOGLEG'
+    bw = resqpy.well.BlockedWell(model,
+                                 well_name = well_name,
+                                 use_face_centres = True,
+                                 grid = grid,
+                                 add_wellspec_properties = True)
+    col_list_orig = ['IW', 'JW', 'L']
+
+    # --------- Act ----------
+    result = bw._BlockedWell__derive_from_wellspec_check_grid_name(check_grid_name = check_grid_name,
```

Comment:
Is this the syntax for calling @staticmethods? Maybe functions like this shouldn't be part of the class at all. I would be inclined to take it out of the BlockedWell class and stick it at the end of the module as a normal standalone function. But others might have other opinions.

Reply:
Agree with @andy-beer here. I think all of the derive_from_wellspec... methods could be converted to functions and put into their own file.